### ISSUE TITLE
bounding boxes are not drawn in the testing  image ,please give me the solution 

### ISSUE THREAD

**Issue by devendraswamy** [RockyXu66/Faster_RCNN_for_Open_Images_Dataset_Keras#36]

**Comment by Parikshit22** [comment]
did you find the solution??
I'm facing the same issue.

**Comment by RaghvendraPal** [comment]
Has anyone found the solution, I am facing the same issue?

**Comment by devendraswamy** [comment]
I trained the model with more epochs and after I get the results

**Comment by RaghvendraPal** [comment]
I have trained 113 epoch, How many epoch have you trained?

**Comment by devendraswamy** [comment]
I tried with 1000 epochs

**Comment by RaghvendraPal** [comment]
epoch_length = 1000
num_epochs = 60
iter_num = 0

which should be 1000?

**Comment by devendraswamy** [comment]
Yup correct

**Comment by devendraswamy** [comment]
No of epochs =1000

**Comment by RaghvendraPal** [comment]
what TensorFlow version have you tried because I am using TensorFlow 2.0

**Comment by RaghvendraPal** [comment]
@Parikshit22 have you got the solution for this problem?

**Comment by RaghvendraPal** [comment]
Hi @devendraswamy could you please share your code, so that I can check hyperparameters.

**Comment by PeymanBaghdadi** [comment]
Hi, I have the same problem, the problem is from :
np.argmax(P_cls[0, ii, :]) == (P_cls.shape[2] - 1)
all the times left side is 3 and right side is 3 too (means 3 == 3) so It does not draw boundingbox.
My config is num_of_epoch = 50 and epoch_length = 1000
I also want to try num_of_epoch = 100 and epoch_length = 1000 .
In some solutions everyone said that if we reduce overlap_thresh it is ok. I do it also but it gives awkward and dummy results.

**Comment by songsonha** [comment]
I recommend you all to use another repo because this issue was fixed for a long time.
For example, I used this repo https://github.com/kentaroy47/frcnn-from-scratch-with-keras
and got no problem.

**Comment by songsonha** [comment]
UPDATEEEEEE: I don't know why results were not drawned on test images. But it happend only in jupyter notebook. I tried to write results with detections to a result folder and then i can see bounding boxes on test images. Just add following codes after the line `print('Elapsed time = {}'.format(time.time() - st))`. Then you can see test images and predicted bounding boxes in results folder. 
Good luck! :)
```import os
if not os.path.isdir("results"):
    os.mkdir("results")
cv2.imwrite('./results/{}.png'.format(idx),img)

**Comment by payamfirouzi** [comment]
I do face the same problem with very bizarre situation, it run perfectly on colab it shows bounding boxes, then I set it to run on cluster but nothing will be shown, even when I set it to train over 1000epochs, nothing will be shown

**Comment by mehnaz1985** [comment]
@songsonha , I have tried but still the same issue. could you please share the complete module? 
print('Elapsed time = {}'.format(time.time() - st)) to the last of that particular part with proper indent? thanks

**Comment by devendraswamy** [comment]
Please use any another repo , this repo for basic understanding but i
should get proper results for after more no of epochs training which its
require a lot of time and lot computation power.

So try once Jwyang faster rcnn repo

**Comment by mehnaz1985** [comment]
@songsonha , thanks for your advice, but I have little knowledge of PyTorch implementation. could you pls send me a tensorflow/keras implementation of faster rcnn? or share your work if possible? and another thing if I would like to use pascal VOC or ms coco how could I try?
As it seems you have already implement faster rcnn that's why asking.

**Comment by devendraswamy** [comment]
Yes u can find the tensorflow faster rcnn on that jwyang repo, his
mentioned in readme please check it once.

**Comment by mehnaz1985** [comment]
Thanks


Mehnaz Tabassum
PhD Candidate
School of SEIT
Federation University, Gippsland Campus
Victoria, Australia.

**Comment by mehnaz1985** [comment]
could you pls explain why i am getting this result
[image: image.png]

Mehnaz Tabassum
PhD Candidate
School of SEIT
Federation University, Gippsland Campus
Victoria, Australia.

**Comment by mehnaz1985** [comment]
![image](https://user-images.githubusercontent.com/66414863/113509660-2bf65c00-959a-11eb-986d-8592eda9850d.png)

**Comment by devendraswamy** [comment]
It seems like the model was overfitted , what is the learning rate ?
Please concentrate on your learning rate if it is high then reduce a little
bit and keep training again , Thank you .

**Comment by mehnaz1985** [comment]
Could u pls indicate the learning rate in the code? I didn't find that. do
you mean the epoch? pls paste that block where i could change the learning
rate. thanks a lot for your time and patient.

Mehnaz Tabassum
PhD Candidate
School of SEIT
Federation University, Gippsland Campus
Victoria, Australia.

**Comment by devendraswamy** [comment]
optimizer = Adam(lr=1e-5) , lr means learning rate , by using
optimizer we can get the global minima in model training so for that
we are using optimizer and we can specify the learning rate for
optimization . depends upon lr, the model will learn with respect to
time and movementum for reaching global minima.


see here lr is 0.00001 so its to high and its leads to model
underfitting,you have smaller data so you can kept smaller lr value ,
lets try with 0.001 and see the results ones,Thank you
